### DISCUSSION TITLE
Decrypt PDF with Just The User Password

### DISCUSSION THREAD

**Discussion by viveksuraiya** [LibrePDF/OpenPDF#835] [Q&A] [answered]
Is it possible to decrypt a PDF with just the user password and not the owner password?
The code I'm using is:

<code>
var reader = new PdfReader("./Nippon.pdf" , 
                    pw.getBytes());

var stamper = new PdfStamper(reader ,
                    new FileOutputStream("./Nippon_e.pdf"));

stamper.close();
</code>
where `pw` is a `String` that holds the password. This works for most pdfs, except from one source.

**Comment by mkl-public** [comment, marked as answer]
Either password should do.

If you have a PDF you have both passwords for and OpenPDF can only read it with the owner password, please share the PDF for analysis.

(One aside, though: `pw.getBytes()` is not necessarily the correct way to get the bytes of a password. If your document uses AES256 (R=6) encryption, the specification requires "The UTF-8 password string shall be generated from Unicode input by processing the input string with the SASLprep (Internet RFC 4013) profile of stringprep (Internet RFC 3454) using the Normalize and BiDi options, and then converting to a UTF-8 representation." In complex cases this may result in something different than your `pw.getBytes()`.)

**Reply by asturio** [reply]
Just to stress that: you shall not use `String.getBytes()` without an encoding, as the result may vary depending of the encoding you the platform running the JVM. And yes, I know there is a lot of that in OpenPDF. Every such call should be checked and "corrected" to use a defined encoding/charset.
